controller/array_action/array_mediator_svc.py

```diff
@@ -1377,3 +1402,90 @@ def _add_snapshot(self, snapshot_name, source_cli_volume, pool):
         if cli_snapshot is None:
             raise array_errors.ObjectNotFoundError(snapshot_id)
         return cli_snapshot
+
+    def _mkvolumegroup(self, name, pool, io_group, source_id):
+        cli_kwargs = {
+            'type': 'clone',
+            'fromsnapshotid': source_id,
+            'pool': pool,
+            'name': name
+        }
+        if io_group:
+            cli_kwargs['iogroup'] = io_group
+        try:
+            return self.client.svctask.mkvolumegroup(**cli_kwargs)
+        except (svc_errors.CommandExecutionError, CLIFailureError) as ex:
+            if is_warning_message(ex.my_message):
+                logger.warning("exception encountered during creation of volume {0}: {1}".format(name, ex.my_message))
+            else:
+                logger.error("Cannot create volume {0}, Reason is: {1}".format(name, ex.my_message))
+                if OBJ_ALREADY_EXIST in ex.my_message:
+                    raise array_errors.VolumeAlreadyExists(name, self.endpoint)
+                if NAME_NOT_EXIST_OR_MEET_RULES in ex.my_message:
+                    raise array_errors.PoolDoesNotExist(pool, self.endpoint)
```

Comment:
how do you know it's the pool and not e.g. the io group?

Reply:
done